lib/logstash/inputs/tcp.rb

```diff
@@ -30,6 +30,10 @@ class LogStash::Inputs::Tcp < LogStash::Inputs::Base
   # `client` connects to a server.
   config :mode, :validate => ["server", "client"], :default => "server"
 
+  # Proxy protocol version, only 1 is supported at this time
+  # http://www.haproxy.org/download/1.5/doc/proxy-protocol.txt
+  config :proxy_protocol, :validate => :number, :default => -1
```

Comment:
I left some comments on the main PR about possibly removing the need for this setting entirely or an alternative of making it a boolean.